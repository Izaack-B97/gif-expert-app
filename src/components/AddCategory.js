import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        // console.log(e.target.value);
        setInputValue(e.target.value);
    };
    
    const handleSubmit = (e) => {
        e.preventDefault(); // Cancela el comportamiento por default
        
        if(inputValue.trim().length > 2) {
            setCategories(categories => [ inputValue, ...categories ]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={ handleSubmit }>
            <h2>AddCategory</h2>
            <input type="text" value={ inputValue } onChange={ handleInputChange }/>
        </form>
    )
}

// Props requeridas
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
