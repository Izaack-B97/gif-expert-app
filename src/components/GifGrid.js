import React, { Fragment } from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
        
    const { data: images ,loading } = useFetchGifs(category);
    // console.log(loading);

    return (
        <Fragment>
            <h3 className="animate__animated animate__fadeIn">{ category }</h3>
            { loading && <p className="animate__animated animate__flash">Loading . . .</p> }
            <div className="card-grid">
                {
                    images.map(image => {
                        return <GifGridItem key={ image.id }{ ...image }/>
                    })
                }
            </div>
        </Fragment>
    )
}

GifGrid.propType = {
    category: PropTypes.string.isRequired
}