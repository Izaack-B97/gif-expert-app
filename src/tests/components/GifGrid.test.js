import '@testing-library/jest-dom';

import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs.js";

jest.mock('../../hooks/useFetchGifs.js'); // Sirve para falsear informacion

// FIXME: Mock; Es fingir algo, falser informacion para que el componente crea que lo tenga

describe('Pruebas en <GifGrid />'  , () => {

    test('debe de renderizar correctamente cuando se carga por primera vez ', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const category = 'Spider man';
        const wrapper = shallow( <GifGrid category={ category }/> );

        expect( wrapper ).toMatchSnapshot();
        
    })
    
    test('debe de mostrar items cuando se cargan las imagenes useFetchGifs ', () => {

        const gifs = [
            {
                id: 125,
                title: 'ABC', 
                url: 'https://www.gayo.com/imagem.png'
            }
        ];  
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const category = 'Spider man';
        const wrapper = shallow( <GifGrid category={ category }/> );

        expect( wrapper ).toMatchSnapshot();
        // Evaluacion de que existen elementos
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    })
    
});