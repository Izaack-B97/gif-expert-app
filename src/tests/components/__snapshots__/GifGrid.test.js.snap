// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan las imagenes useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Spider man
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={125}
      key="125"
      title="ABC"
      url="https://www.gayo.com/imagem.png"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de renderizar correctamente cuando se carga por primera vez  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Spider man
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading . . .
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
